Snakefile

```diff
@@ -8,6 +8,69 @@ wildcard_constraints:
 SEGMENTS = ["pb2", "pb1", "pa", "ha","np", "na", "mp", "ns"]
 #SUBTYPES = ["h5n1", "h5nx", "h7n9", "h9n2"]
 
+CURRENT_BASEDIR = workflow.current_basedir # TODO XXX store this value here - can't access within functions because workflow.included_stack is empty
+
+# Load the base config.yaml relative to the entry snakefile (i.e. not this snakefile)
+if os.path.exists(os.path.join(workflow.basedir, 'config.yaml')):
+    configfile: os.path.join(workflow.basedir, 'config.yaml')
+
+# load a config.yaml file if it exists in the current working directory
+if os.path.exists("config.yaml"):
+    configfile: "config.yaml"
+
+from pprint import pp; pp(config, stream=sys.stderr) # TODO XXX remove
+
+class InvalidConfigError(Exception):
+    pass
+
+def resolve_config_path(original_path, wildcards=None):
+    """
+    Resolve a relative *path* given in a configuration value. Before resolving
+    any '{x}' substrings are replaced by their corresponding wildcards (if the
+    `wildcards` argument is provided).
+    
+    Search order (first match returned):
+    1. Relative to the analysis directory
+    2. Relative to the directory the entry snakefile was in. Typically this
+       is not the Snakefile you are looking at now but (e.g.) the one in
+       avian-flu/gisaid
+    3. Relative to where this Snakefile is (i.e. `avian-flu/`)
+    """
+    path = original_path.format(**wildcards) if wildcards else original_path
+
+    if re.search(r'\{.+\}', path):
+        if wildcards:
+            print(f"The call to `resolve_config_path({original_path!r}, wildcards)` still includes wildcard-like placeholders afrer resolving: {path!r}.", file=sys.stderr)
+        else:
+            print(f"The call to `resolve_config_path({original_path!r})` includes unresolved wildcards - please include the wildcards as the second argument to `resolve_config_path`.", file=sys.stderr)
+        exit(2)
+
+    if os.path.exists(path): # isfile?
+        return path
+
+    # Check if the path exists relative to the subdir where the entry snakefile is
+    # (e.g. avian-flu/gisaid). If you want to use further subdirectories (e.g. avian-flu/gisaid/config/x.tsv)
+    # you're expected to supply the 'config/x.tsv' as the value in the config YAML
+    # NOTE: this means analysis directory overrides have to use that same 'config/x.tsv' structure, but
+    # given the different directories avian-flu uses that's acceptable. In other words, if we standardised
+    # avian-flu then we could add subdirectories to the search order here
+    basepath = os.path.join(workflow.basedir, path)
```

Comment:
I don't imagine this is going to work with docker / AWS etc as it produces absolute paths, e.g. `--sequences /Users/naboo/github/nextstrain/avian-flu/ingest/fauna/results/sequences_ha.fasta`. I presume this is a decision which involves how/where the nextstrain CLI wants to bind the various directories involved here

Reply:
Absolute paths should be just fine in containerized runtimes, as long as they're dynamically resolved/generated from relative paths, as in here.